presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImageListPresenter.java

```diff
@@ -0,0 +1,81 @@
+package com.globant.equattrocchio.cleanarchitecture.mvp.presenter;
+
+import android.app.Activity;
+
+import com.globant.equattrocchio.cleanarchitecture.mvp.view.imageList.ImageListView;
+import com.globant.equattrocchio.cleanarchitecture.util.bus.RxBus;
+import com.globant.equattrocchio.cleanarchitecture.util.bus.observers.ImageDetailButtonObserver;
+import com.globant.equattrocchio.cleanarchitecture.util.bus.observers.LatestImagesButtonObserver;
+import com.globant.equattrocchio.data.ImagesServicesImpl;
+import com.globant.equattrocchio.domain.model.ImageEntity;
+import com.globant.equattrocchio.domain.useCases.GetLatestImagesUseCase;
+
+import java.util.List;
+
+import io.reactivex.annotations.NonNull;
+import io.reactivex.observers.DisposableObserver;
+
+public class ImageListPresenter {
+
+    private ImageListView view;
+    private GetLatestImagesUseCase getLatestImagesUseCase;
+
+    public ImageListPresenter(ImageListView view, GetLatestImagesUseCase getLatestImagesUseCase) {
+        this.view = view;
+        this.getLatestImagesUseCase = getLatestImagesUseCase;
+    }
+
+    private void onLatestImageResponse(List<ImageEntity> imageEntities) {
+        view.showImageCards(imageEntities);
+    }
+
+    private void onLatestImagesButtonPressed() {
+        getLatestImagesUseCase.execute(new DisposableObserver<List<ImageEntity>>() {
+            @Override
+            public void onNext(@NonNull List<ImageEntity> imageEntities) {
+                onLatestImageResponse(imageEntities);
+            }
+
+            @Override
+            public void onError(@NonNull Throwable e) {
+                view.showError();
+            }
+
+            @Override
+            public void onComplete() {
+                new ImagesServicesImpl().getLatestImages(null);
```

Comment:
why are you calling here the ImagesServicesImpl()? you should call `view.hideLoader()` in order to show the user that the app finished the work.